TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+
+## What is a Token?
+Tokens are a form of authentication that verifies whether the user can access parts of an application. In this case, we want to use them to authenticate the user and their privileges within a video call. 
+
+The token is generated based on the HMAC-SHA256 cryptographic algorithm. This algorithm is an industry standard for authentication. 
+
+We don't need to dive into cybersecurity theory; just know that it's a good algorithm that will give you a safe token, and we can generate this token using the `agora-token` package.
+
+## Setting up an Endpoint in Astro
+To create an endpoint in Astro, you need to add a Javascript or Typescript file to the `pages` directory. This file should have the file name, followed by the return datatype extension, followed by `.js` or `.ts`.The Javascript and Typescript extensions will be removed during the build process.
+
+The most common type of return data is JSON, so our file will be a `.json.ts` file. We also need to pass some parameters. In Astro, you do this using brackets for the file/folder name. 
+
+Our file path for generating tokens will be `pages/rtc/[channel]/[role]/[uid].json.ts`, which will result in the end point of `rtc/{channel}/{role}/{uid}.json`.
+
+## Define the endpoint
+Our endpoint will be a `GET` request because our objective is to retrieve a token. You can define this in Astro by exporting a `GET` function and returning a JSON Response.
+
+We also import the `agora-token` package and bring in our `APP_ID` and `APP_CERTIFICATE`, which we defined in the `.env` file.
+
+
+```ts
+import agoraToken from "agora-token";
+import type { APIContext } from "astro";
+
+const APP_ID = import.meta.env.APP_ID;
+const APP_CERTIFICATE = import.meta.env.APP_CERTIFICATE;
+
+export async function GET({ params }: APIContext) {
+  return new Response(
+    JSON.stringify({
+      rtcToken: "token",
+    })
+  )
+}
+```
+
+## CORS
+We first need to set up our headers so that we don't encounter any CORS issues when using our backend.
+
+```ts
+export async function GET({ params }: APIContext) {
+    const headers = new Headers({
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": "GET, OPTIONS",
+    });
+
+    return new Response(JSON.stringify({
+        rtcToken: "token"
+    }), { headers })
+}
+```
+
+* "Access-Control-Allow-Origin": "*" allows all domains to access the server, which is helpful for APIs or during development.
+* "Access-Control-Allow-Methods": "GET, OPTIONS" specifies that only GET and OPTIONS methods are permitted, controlling how other sites can interact with the server.
+
+These headers will allow our API to work well in production and development, but for production you should limit only the origins that should be allowed.
+
+## Input Checks
+To generate the token, we need a few input parameters that we retrieve from the route. We need to check that `channel` and `uid` are not empty. If they are, return a Bad Request Error.
```

Comment:
Consider re-wording this sentence to explain why we need the parameters.

Reply:
Updated to:

> The token generation function requires the `channel`, `role`, and `uid`. These are used to ensure that the token includes correct authentication and privileges. To ensure we pass accurate data, check that `channel` and `uid` are not empty. If they are, return a Bad Request Error.